cmd/podman/machine/client9p.go

```diff
@@ -69,8 +70,25 @@ func client9p(portNum uint32, mountPath string) error {
 
 	logrus.Infof("Going to mount 9p on vsock port %d to directory %s", portNum, mountPath)
 
-	// Host connects to non-hypervisor processes on the host running the VM.
-	conn, err := vsock.Dial(vsock.Host, portNum, nil)
+	// The server is starting at the same time.
+	// Perform up to 5 retries with a backoff.
+	var (
+		conn       *vsock.Conn
+		retries    = 20
+	)
+	for i := 0; i < retries; i++ {
+		// Host connects to non-hypervisor processes on the host running the VM.
+		conn, err = vsock.Dial(vsock.Host, portNum, nil)
+		if err == nil {
+			break
+		}
+		// If errors.Is worked on this error, we could detect non-timeout errors.
+		// But it doesn't. So retry 5 times regardless.
+		if err == nil {
+			break
+		}
```

Comment:
weird, reading the vsock code I see no reason why errors.Is shouldn't work.

Anyhow you now have the err == nil check twice so it should be dropped once here.